x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -9,24 +9,72 @@ package kubernetes
 
 import (
 	"time"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 // Config for kubernetes provider
 type Config struct {
+	Scope     string    `config:"scope"`
+	Resources Resources `config:"resources"`
+
+	// expand config settings at root level of config too
 	KubeConfig     string        `config:"kube_config"`
+	Namespace      string        `config:"namespace"`
 	SyncPeriod     time.Duration `config:"sync_period"`
 	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
 
-	// Needed when resource is a pod
+	// Needed when resource is a Pod or Node
 	Node string `config:"node"`
+}
 
-	// Scope of the provider (cluster or node)
-	Scope string `config:"scope"`
+// Resources config section for resources' config blocks
+type Resources struct {
+	Pod     *ResourceConfig `config:"pod"`
+	Node    *ResourceConfig `config:"node"`
+	Service *ResourceConfig `config:"service"`
+}
+
+// ResourceConfig for kubernetes resource
+type ResourceConfig struct {
+	KubeConfig     string        `config:"kube_config"`
+	Namespace      string        `config:"namespace"`
+	SyncPeriod     time.Duration `config:"sync_period"`
+	CleanupTimeout time.Duration `config:"cleanup_timeout" validate:"positive"`
+
+	// Needed when resource is a Pod or Node
+	Node string `config:"node"`
```

Comment:
I'm wondering about use cases for resource specific settings, do you have anyone in mind?

Reply:
Now that I'm thinking of it again, maybe it is over-engineering to provide this option at the moment since the base config shared for all the resources should cover the cases. Flexibility for different accesses per resource or variant settings options would be nice to think of but we can and see if users actually need them. So, I will change it and move to single config for all of the resources.